dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/AuthorizationRestRepository.java

```diff
@@ -0,0 +1,268 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.dspace.app.rest.Parameter;
+import org.dspace.app.rest.SearchRestMethod;
+import org.dspace.app.rest.authorization.Authorization;
+import org.dspace.app.rest.authorization.AuthorizationFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeatureService;
+import org.dspace.app.rest.authorization.AuthorizationRestUtil;
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.model.AuthorizationRest;
+import org.dspace.app.rest.model.BaseObjectRest;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.service.EPersonService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+
+/**
+ * This is the repository responsible to manage Authorization Rest object
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ */
+
+@Component(AuthorizationRest.CATEGORY + "." + AuthorizationRest.NAME)
+public class AuthorizationRestRepository extends DSpaceRestRepository<AuthorizationRest, String> {
+
+    private static final Logger log = LoggerFactory.getLogger(AuthorizationRestRepository.class);
+
+    @Autowired
+    private AuthorizationFeatureService authorizationFeatureService;
+
+    @Autowired
+    private AuthorizationRestUtil authorizationRestUtil;
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    @Autowired
+    private EPersonService epersonService;
+
+    @Autowired
+    protected ConverterService converter;
+
+    @Override
+    @PreAuthorize("hasPermission(#id, 'authorization', 'READ')")
+    public AuthorizationRest findOne(Context context, String id) {
+
+        AuthorizationRest authorizationRest = null;
+
+        String featureName;
+        try {
+            featureName = authorizationRestUtil.getFeatureName(id);
+        } catch (IllegalArgumentException e) {
+            log.warn(e.getMessage(), e);
```

Comment:
Perhaps add a small message here for the log ?

Reply:
it is provided by the getFeatureName inside the IllegalArgumentException. For such reason the exception is stated in the method javadoc despite to be a runtime exception